From b019a51d3948c95f10df73651b16882e4efceb95 Mon Sep 17 00:00:00 2001
From: Arulanbu Balusamy <quic_abalusam@quicinc.com>
Date: Thu, 10 Oct 2024 10:52:30 +0530
Subject: [PATCH] QSDK:PROTOCOL:HOSTAPD:KW:hostapd:Fix KW issue to avoid banned function.

As 'strtok' has been deprecated and it is not safe, replace it
with a safe function 'strtok_r'

Signed-off-by: Arulanbu Balusamy <quic_abalusam@quicinc.com>
---
 hostapd/ctrl_iface.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/hostapd/ctrl_iface.c b/hostapd/ctrl_iface.c
index 20f5cfe..2207a2a 100644
--- a/hostapd/ctrl_iface.c
+++ b/hostapd/ctrl_iface.c
@@ -162,15 +162,15 @@ static char *get_option(char *opt, char *str)
 static struct hostapd_config *hostapd_ctrl_iface_config_read(const char *fname)
 {
 	struct hostapd_config *conf;
-	char *opt, *val;
+	char *opt, *val, *sptr;
 
 	conf = hostapd_config_read(fname);
 	if (!conf)
 		return NULL;
 
-	for (opt = strtok(reload_opts, " ");
+	for (opt = strtok_r(reload_opts, " ", &sptr);
 	     opt;
-		 opt = strtok(NULL, " ")) {
+		 opt = strtok_r(NULL, " ", &sptr)) {
 
 		if ((val = get_option(opt, "channel=")))
 			conf->channel = atoi(val);
-- 
2.34.1

